### PR TITLE
[eclipse/xtext#1897] converted xtend code to java

### PR DESCRIPTION
**Pull request by cdietrich**
[eclipse/xtext#1897] converted xtend code to java
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/editor/quickfix/CompositeQuickfixTest.java [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+/**
+ * @author Moritz Eysholdt - Initial contribution and API
+ */
+@RunWith(XtextRunner.class)
+@InjectWith(QuickfixCrossrefTestLanguageUiInjectorProvider.class)
+public class CompositeQuickfixTest extends AbstractMultiQuickfixTest {
+	@Test
+	public void testSimpleFixMultipleMarkers() throws Exception {
+		String model =
+				"\"bad doc\"\n" +
```
Comment (ArneDeutsch) [inline org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/editor/quickfix/CompositeQuickfixTest.java:43]: Have you checked that the tests succeed on windows, too?

File: org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/editor/quickfix/CompositeQuickfixTest.java [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+		IFile resource = dslFile(model);
+		IMarker[] markers = getMarkers(resource);
+		String contentAndMarkersBefore =
+				"<0<c>0> {	badname { foo {} } }\n" +
+				"<1<a>1> {	badname { bar {} } }\n" +
+				"-----\n" +
+				"0: message=badNameInSubelements\n" +
+				"1: message=badNameInSubelements\n";
+		assertContentsAndMarkers(resource, markers, contentAndMarkersBefore);
+		IMarker firstMarker = Iterables.getFirst(IterableExtensions.sortBy(Arrays.asList(markers), (IMarker it) -> {
```
Comment (ArneDeutsch) [inline org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/editor/quickfix/CompositeQuickfixTest.java:167]: Second instance of this code ... probably possible to extract a helper method!?

File: org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/editor/quickfix/CompositeQuickfixTest.java [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+
+	@Test
+	public void testMultiQuickAssist() throws Exception {
+		String model =
+				"c {	badname { foo {} } }\n" +
+				"a {	badname { bar {} } }\n";
+		IFile resource = dslFile(model);
+		XtextEditor editor = openEditor(resource);
+		ICompletionProposal[] proposals = computeQuickAssistProposals(editor, 1);
+		Assert.assertEquals("Fix Bad Names".toString(),
```
Comment (ArneDeutsch) [inline org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/editor/quickfix/CompositeQuickfixTest.java:191]: Static import for Assert would make the code slightly more readable.

File: org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/editor/quickfix/MultiQuickFixTest.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+		IMarker[] markers = getMarkers(resource);
+		String contentAndMarkersBefore =
+				"<0<\"no doc\">0>\n" +
+				"Foo { ref Bor }\n" +
+				"<1<\"no doc\">1> Bor { }\n" +
+				"-----\n" +
+				"0: message=multiFixableIssue\n" +
+				"1: message=multiFixableIssue\n";
+		assertContentsAndMarkers(resource, markers, contentAndMarkersBefore);
+		IMarker firstMarker = Iterables.getFirst(IterableExtensions.sortBy(Arrays.asList(markers), (IMarker it) -> {
```
Comment (ArneDeutsch) [inline org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/editor/quickfix/MultiQuickFixTest.java:80]: Code is repeated accross test files ... a shared helper method would be nice.

File: org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/editor/quickfix/MultiQuickFixTest.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+		ICompletionProposal[] proposals = computeQuickAssistProposals(editor, 1);
+		Assert.assertEquals("Multi fix".toString(),
+				Joiner.on("\n").join(Lists.transform(Arrays.asList(proposals), ICompletionProposal::getDisplayString)));
+		proposals[0].apply(editor.getDocument());
+		String expectation =
+				"\n" +
+				"\"Better documentation\"\n" +
+				"Foo { ref Bor }\n" +
+				"\"no doc\" Bor { }\n";
+		Assert.assertEquals(expectation.toString(), editor.getDocument().get());
```
Comment (ArneDeutsch) [inline org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/editor/quickfix/MultiQuickFixTest.java:115]: Static import.

File: org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/xmleditor/XmlParsingTest.java [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+	@Inject
+	private ParseHelper<XmlDocument> parseHelper;
+
+	@Inject
+	private ValidationTestHelper validationTestHelper;
+
+	@Test
+	public void test001() throws Exception {
+		XmlDocument model = parseHelper.parse("");
+		validationTestHelper.assertNoErrors(model);
```
Comment (ArneDeutsch) [inline org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/xmleditor/XmlParsingTest.java:34]: Create a helper method "assertNoErrors" that does the parsing and the delegation to simplify the code to "assertNoErrors("")".

File: org.eclipse.xtext.xtext.ui.tests/src/org/eclipse/xtext/xtext/ui/editor/folding/XtextFoldingTest.java [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+ * @author miklossy - Initial contribution and API
+ */
+@RunWith(XtextRunner.class)
+@InjectWith(XtextUiInjectorProvider.class)
+public class XtextFoldingTest extends AbstractFoldingTest {
+	@Test
+	public void greetings() {
+		String model =
+				"grammar org.xtext.example.mydsl.MyDsl with org.eclipse.xtext.common.Terminals\n" +
+				"\n" +
```
Comment (ArneDeutsch) [inline org.eclipse.xtext.xtext.ui.tests/src/org/eclipse/xtext/xtext/ui/editor/folding/XtextFoldingTest.java:28]: Check that the newline isn't an issue for windows machines!?

**Review by ArneDeutsch — CHANGES_REQUESTED** [review]

File: org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/editor/quickfix/CompositeQuickfixTest.java (hunk 1 above)
Comment (cdietrich) [reply org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/editor/quickfix/CompositeQuickfixTest.java:43]: yes

File: org.eclipse.xtext.xtext.ui.tests/src/org/eclipse/xtext/xtext/ui/editor/folding/XtextFoldingTest.java (hunk 7 above)
Comment (cdietrich) [reply org.eclipse.xtext.xtext.ui.tests/src/org/eclipse/xtext/xtext/ui/editor/folding/XtextFoldingTest.java:28]: no that works

File: org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/xmleditor/XmlParsingTest.java (hunk 6 above)
Comment (cdietrich) [reply org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/xmleditor/XmlParsingTest.java:34]: dont want to start a big refactoring

File: org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/editor/quickfix/MultiQuickFixTest.java (hunk 4 above)
Comment (cdietrich) [reply org.eclipse.xtext.ui.tests/tests/org/eclipse/xtext/ui/tests/editor/quickfix/MultiQuickFixTest.java]: done

**Review by ArneDeutsch — APPROVED** [review]

**Pull request merged**